### PR TITLE
Feature: Add volume slider to remote screen

### PR DESCRIPTION
**Pull request by wutschel**
## Description

Closes https://github.com/xbmc/Official-Kodi-Remote-iOS/issues/358.

Add volume slider to remote screen:
- When screen size allows (`height >= 568`) add the volume bar on top.
- Use `leftAnchor` to differentiate between embedded and full screen.
- Align both iPad/iPhone volume elements in vertical center of screen.

## Summary for release notes

Feature: Show volume slider on remote screen (not working for iPhone 4S)

### CODE REVIEW

File: XBMC Remote/RemoteController.m [hunk 1]
```diff
@@ -193,11 +193,19 @@ - (void)configureView {
...
         CGRect frame = remoteControlView.frame;
         frame.size.height *= transform;
         frame.size.width *= transform;
         frame.origin.y = [self getOriginYForRemote:remoteControlView.frame.size.height - frame.size.height - toolbarPadding];
+        
+        if ([Utilities hasRemoteToolBar]) {
+            volumeSliderView = [[VolumeSliderView alloc] initWithFrame:CGRectZero leftAnchor:0.0];
+            [volumeSliderView startTimer];
+            [self.view addSubview:volumeSliderView];
+            frame.origin.y = frame.origin.y == 0 ? volumeSliderView.frame.size.height : frame.origin.y;
```
Comment (kambala-decapitator) [inline XBMC Remote/RemoteController.m:207]: Use if to prevent useless assignment

File: XBMC Remote/VolumeSliderView.m [hunk 2]
```diff
@@ -105,6 +99,11 @@ - (id)initWithFrame:(CGRect)frame {
             muteBackgroundImage = [Utilities colorizeImage:img withColor:[UIColor darkGrayColor]];
             volumeIconColor = [UIColor grayColor];
         }
+        // Move all buttons to vertical center of view
+        for (UIView *subView in [self subviews]) {
+            subView.center = CGPointMake(subView.center.x, self.frame.size.height / 2);
```
Comment (kambala-decapitator) [inline XBMC Remote/VolumeSliderView.m:104]: 1. self.subviews
2. Pre-compute y coordinate outside the loop (also CGRectGetHeight)

**Review by kambala-decapitator — CHANGES_REQUESTED** [review]

File: XBMC Remote/RemoteController.m (hunk 1 above)
Comment (wutschel) [reply XBMC Remote/RemoteController.m]: Done

File: XBMC Remote/VolumeSliderView.m (hunk 2 above)
Comment (wutschel) [reply XBMC Remote/VolumeSliderView.m]: I hope I got the 2nd comment right.

**Comment by wutschel** [comment]
I still still have a (minor) position glitch I would like to solve, and will keep this draft until I found a good solution.

**Comment by wutschel** [comment]
> I still still have a (minor) position glitch I would like to solve, and will keep this draft until I found a good solution.

Solved now and added as first commit -- this change would also be correct without the changes to add the volume slider to the remote screen. I also tested with setting `hasRemoteToolBar` to `false` to simulate iPhone 4S screen size. This seems to work as expected. I am marking this as ready for review.

File: XBMC Remote/RemoteController.m [hunk 3]
```diff
@@ -113,7 +114,7 @@ - (CGFloat)getOriginYForRemote:(CGFloat)offsetBottomMode {
 - (void)setEmbeddedView {
     CGRect frame = TransitionalView.frame;
     CGFloat newWidth = CGRectGetWidth(UIScreen.mainScreen.fixedCoordinateSpace.bounds) - ANCHOR_RIGHT_PEEK;
-    CGFloat shift;
+    CGFloat shift = 0;
```
Comment (kambala-decapitator) [inline XBMC Remote/RemoteController.m:117]: Looks useless as value is assigned in both cases

File: XBMC Remote/RightMenuViewController.m [hunk 4]
```diff
@@ -44,6 +44,9 @@ - (CGFloat)tableView:(UITableView*)tableView heightForRowAtIndexPath:(NSIndexPat
     else if ([tableData[indexPath.row][@"label"] isEqualToString:@"RemoteControl"]) {
         return UIScreen.mainScreen.bounds.size.height - [self getRemoteViewOffsetY];
     }
+    else if ([tableData[indexPath.row][@"label"] isEqualToString:@"VolumeControl"]) {
```
Comment (kambala-decapitator) [inline XBMC Remote/RightMenuViewController.m:47]: I strongly suggest to extract the left part into a variable

**Review by kambala-decapitator — CHANGES_REQUESTED** [review]
Everything else is fine

File: XBMC Remote/RemoteController.m (hunk 3 above)
Comment (wutschel) [reply XBMC Remote/RemoteController.m:117]: True, made this change while commenting the whole block which hides/modes buttons. Will be removed with next forced push (not worth a fixup-commit).

File: XBMC Remote/RightMenuViewController.m (hunk 4 above)
Comment (wutschel) [reply XBMC Remote/RightMenuViewController.m]: Done

**Comment by wutschel** [comment]
Last fixup provided.

**Review by kambala-decapitator — APPROVED** [review]

**Comment by wutschel** [comment]
Here we go.

**Pull request merged**